app/app.go

```diff
@@ -1294,32 +1294,49 @@ func (d *Deployments) GetDeploymentForDeviceWithCurrent(ctx context.Context, dev
 		return nil, nil
 	}
 
-	if installed.ArtifactName != "" && deployment.ArtifactName == installed.ArtifactName {
-		// if the device reported same artifact name as the one in the device deployment,
-		// and this is a new device deployment - indicated by device deployment
-		// status "pending",
-		// pretend there is no deployment for this device, but update
-		// its status to already installed first
-		if deviceDeployment.Status == model.DeviceDeploymentStatusPending {
+	if deviceDeployment.Request != nil &&
+		(deviceDeployment.Request.DeviceType != installed.DeviceType ||
+			deviceDeployment.Request.ArtifactName != installed.ArtifactName) {
```

Comment:
nit: This can be simplified a little:
```suggestion
	if deviceDeployment.Request != nil && !reflect.DeepEqual(deviceDeployment.Request, installed) {
```

Reply:
as you wish!